packages/tabs/addon/components/nucleus-tabs/tab-list-item.js

```diff
@@ -0,0 +1,320 @@
+  *
+  * @field disabled
+  * @type string
+  * @default null
+  * @readonly
+  * @public
+  */
+  @defaultProp
+  disabled = null;
+
+  /**
+  * controls
+  *
+  * @field controls
+  * @description idref to what panel this tab item controls
+  * @type string|null
+  * @readonly
+  * @public
+  */
+  @defaultProp
+  controls;
+
+  /**
+  * selected
+  *
+  * @field selected
+  * @description currently selected tab
+  * @readonly
+  * @public
+  */
+  @defaultProp
+  selected;
+
+  /**
+  * tabOrder
+  *
+  * @field tabOrder
+  * @description order in which tabs are displayed. Only first tabs will not have tabIndex value
+  * @type number
+  * @readonly
+  * @public
+  */
+  @defaultProp
+  tabOrder;
+
+  /**
+  * role
+  *
+  * @field role
+  * @type string
+  * @default 'tab'
+  * @public
+  */
+  role = 'tab';
+
+  /**
+  * isPressed
+  *
+  * @field isPressed
+  * @description to solve focus on click styling
+  * @type boolean
+  * @default false
+  * @public
+  */
+  isPressed = false;
+
+  /**
+  * tabindex
+  *
+  * @field tabindex
+  * @type string|null
+  * @public
+  */
+  @computed('tabOrder', function() {
+    let tabIndex = null;
+    if(!get(this, 'isDisabled')) {
+      tabIndex = (get(this, 'tabOrder') === 0)? '0' : '-1';
+    }
+    return tabIndex;
+  })
+  tabindex;
+
+  /**
+  * isActive
+  *
+  * @field isActive
+  * @type boolean
+  * @public
+  */
+  @computed('selected', function() {
+    return (get(this, 'selected') === get(this, 'name'));
+  })
+  isActive;
+
+  /**
+  * isDisabled
+  *
+  * @field isDisabled
+  * @type boolean
+  * @public
+  */
+  @computed('disabled', function() {
+    return (get(this, 'disabled').toString() === 'true')? true : false;
+  })
+  isDisabled;
+
+  /**
+  * title
+  *
+  * @field title
+  * @type string
+  * @public
+  */
+  @computed('name', function() {
+    return get(this, 'name');
+  })
+  title;
+
+  /**
+  * aria-controls
+  *
+  * @field aria-controls
+  * @type string
+  * @public
+  */
+  @computed('controls', function() {
+    return get(this, 'controls');
+  })
+  'aria-controls';
+
+  /**
+  * aria-selected
+  *
+  * @field aria-selected
+  * @type boolean
+  * @public
+  */
+  @computed('selected', function() {
+    return (get(this, 'selected') === get(this, 'name')).toString();
+  })
+  'aria-selected';
+
+  /**
+  * init
+  *
+  * @method init
+  * @description lifecycle event
+  * @public
+  *
+  */
+  init() {
+    super.init(...arguments);
+  }
+
+  /**
+  * didInsertElement
+  *
+  * @method didInsertElement
+  * @description lifecycle event
+  * @public
+  *
+  */
+  didInsertElement() {
+    get(this, 'registerTabListItem').call(this, {
+      id: get(this, 'elementId'),
+      name: get(this, 'name')
+    });
+  }
+
+  /**
+  * mouseDown
+  *
+  * @method mouseDown
+  * @description event handler : We are negating the style applied on click-focus
+  * @public
+  *
+  */
+  mouseDown() {
+    if(get(this, 'isDisabled') === false) set(this, 'isPressed', true);
```

Comment:
Not sure if there's a linter rule for this, but can you wrap it in `{ }`

```js
mouseDown() {
    if(get(this, 'isDisabled') === false) {
       set(this, 'isPressed', true);
    }
}
```

P.S: It's just a convention that we follow in our codebase.

Reply:
Sure, will follow this hereon.